pytorch_lightning/trainer/training_loop.py

```diff
@@ -361,6 +318,8 @@ def _process_training_step_output_1_0(self, training_step_output, split_batch):
         if isinstance(training_step_output, dict):
             loss = training_step_output.pop("loss", None)
             hiddens = training_step_output.pop("hiddens", None)
+            if hiddens is not None:
+                hiddens = hiddens.detach()
```

Comment:
We are detaching here because of truncated backpropagation through time, right?
So was this simply missing before?

Reply:
Yes. I moved it to the wrong place in #6016 so this fixes that unreleased bug